src/App.tsx

```diff
@@ -1,61 +1,112 @@
+import React, { useState } from 'react';
+
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types';
+import { Todo } from './types';
+import { TodoList } from './components/TodoList/';
+
+function getUserById(userId: User['id']) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const initialTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [selectError, setSelectError] = useState(false);
+
+  function addTodo(newTodo: Todo) {
+    const addedTodo: Todo = {
+      ...newTodo,
+      id: Math.max(...todos.map(todo => todo.id)) + 1,
+    };
+
+    setTodos(currTodo => [...currTodo, addedTodo]);
+  }
+
+  function handleSubmit(e: React.FormEvent) {
+    e.preventDefault();
+
+    setTitleError(!title.trim());
+    setSelectError(!selectedUserId);
+
+    if (!title.trim() || !selectedUserId) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: 0,
+      title,
+      completed: false,
+      userId: selectedUserId,
+      user: getUserById(selectedUserId),
+    };
+
+    addTodo(newTodo);
+
+    setTitle('');
+    setSelectedUserId(0);
+  }
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            data-cy="titleInput"
+            value={title}
+            onChange={e => {
+              setTitle(e.target.value);
+              setTitleError(false);
+            }}
+            placeholder="Enter a title"
+          />
+          {titleError && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
+          <select
+            data-cy="userSelect"
+            onChange={e => {
+              setSelectedUserId(+e.target.value);
+              setSelectError(false);
+            }}
```

Comment:
The same issue here

Reply:
Ok. But talking about linter - I cannot see any issues within my scss files. I guess problem pops after compilation